File: src/main/java/com/manning/simplysend/repositories/UserCredentialsRepository.java
package com.manning.simplysend.repositories;

import com.manning.simplysend.entities.UserCredentials;
import org.springframework.data.repository.CrudRepository;

import java.util.Optional;

public interface UserCredentialsRepository extends CrudRepository<UserCredentials, Long> {
    Optional<UserCredentials> findByUser_Email(String email);
}

File: src/main/kotlin/com/manning/simplysend/repositories/UserCredentialsRepository.kt
package com.manning.simplysend.repositories

import com.manning.simplysend.entities.UserCredentials
import org.springframework.data.repository.CrudRepository
import java.util.*

interface UserCredentialsRepository : CrudRepository<UserCredentials, Long> {

    fun findByUser_Email(email: String): Optional<UserCredentials>
}